File: 1496.path-crossing.py
#
# @lc app=leetcode id=1496 lang=python3
#
# [1496] Path Crossing
#

# @lc code=start
class Solution:
    def isPathCrossing(self, path: str) -> bool:
        seen = set()
        seen.add((0,0))
        x, y = 0, 0
        for c in path:
            if c == 'N':
                y += 1
            elif c == 'E':
                x += 1
            elif c == 'S':
                y -= 1
            else:
                x -= 1
            if (x, y) in seen:
                return True
            else:
                seen.add((x,y))
        return False
# @lc code=end

File: 1496.path-crossing.cs
/*
 * @lc app=leetcode id=1496 lang=csharp
 *
 * [1496] Path Crossing
 */

// @lc code=start
public class Solution 
{
    public bool IsPathCrossing(string path) 
    {
        HashSet<Tuple<int, int>> seen = new HashSet<Tuple<int, int>>();
        int x = 0;
        int y = 0;
        seen.Add(new Tuple<int,int> (x, y));
        foreach (char c in path)
        {
            if (c == 'N')
            {
                y++;
            }
            else if (c == 'E')
            {
                x++;
            }
            else if (c == 'S')
            {
                y--;
            }
            else
            {
                x--;
            }
            if (seen.Contains(new Tuple<int,int> (x, y)))
            {
                return true;
            }
            else
            {
                seen.Add(new Tuple<int,int> (x, y));
            }
        }
        return false;  
    }
}
// @lc code=end

